buildtest/buildsystem/base.py

```diff
@@ -208,31 +207,29 @@ class BuilderBase:
        any kind of builder.
     """
 
-    def __init__(self, name, recipe_config, config_file=None, testdir=None):
-        """initiate a builder base. A recipe configuration (loaded) is required.
-           this can be handled easily with the BuildConfig class:
+    def __init__(self, name, recipe, buildspec=None, testdir=None):
+        """Initiate a builder base. A recipe configuration (loaded) is required.
+           this can be handled easily with the BuildspecParser class:
 
-           bc = BuildConfig(config_file)
-           recipe_config = bc.get("section_name")
-           builder = ScriptBuilder(recipe_config)
+           bc = BuildspecParser(buildspec)
+           recipe = bc.get("section_name")
+           builder = ScriptBuilder(recipe)
 
            Parameters:
 
-           name: a name for the build recipe (required)
-           recipe_config: the loaded section from the config_file for the user.
-           config_file: the pull path to the configuration file, must exist.
+           name: a name for the Buildspec recipe (required)
+           recipe: the loaded section from the buildspec for the user.
+           buildspec: the pull path to the Buildspec file, must exist.
         """
 
         self.name = name
         self.pwd = os.getcwd()
         self.result = {}
         self.build_id = None
         self.metadata = {}
-        self.config_file = config_file
-        self.config_name = re.sub("[.](yml|yaml)", "", os.path.basename(config_file))
-        self.testdir = testdir or os.path.join(
-            os.getcwd(), ".buildtest", self.config_name
-        )
+        self.buildspec = buildspec
+        self.name = re.sub("[.](yml|yaml)", "", os.path.basename(buildspec))
```

Comment:
This is not correct - self.name is already set above. This is supposed to be the name of the file, not the name of the key/step.

Reply:
I changed line 231 to use the name ``self.config_name`` i realized that was a mistake on my end, where ``self.name=name``  where name was passed as argument to ``BuilderBase``